### PR TITLE
Creates `IssuedCurrencyClient` and corresponding test file.

### PR DESCRIPTION
**Pull request by amiecorso**
## High Level Overview of Change
This PR creates the skeleton of an `IssuedCurrencyClient` upon which we can begin implementing the required methods for handling Issued Currencies on the XRPL.  This is modeled closely after the `DefaultXrpClient`.  Note that it doesn't implement an interface or a decorator... We're going to begin by implementing methods here, and make larger design choices in the future when it comes time to expose these methods publicly and in conjunction with the existing `XrpClient` methods.

### Context of Change
IOUs gogogo!

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Test Plan
CI

### CODE REVIEW

File: src/XRP/issued-currency-client.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+   * @param network The network this IssuedCurrencyClient is connecting to.
+   */
+  public constructor(
+    private readonly networkClient: XrpNetworkClient,
+    readonly network: XrplNetwork,
+  ) {
+    this.coreXrplClient = new CoreXrplClient(networkClient, network)
+  }
+
+  public shutUpCompiler(): void {
```
Comment (tedkalaw) [inline src/XRP/issued-currency-client.ts:49]: it's getting mad bc this stuff is unused right? you could use `/* eslint-disable no-unused-vars */ ` instead, although i expect that these are going to get used pretty much immediately

File: test/XRP/issued-currency-client.test.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { XrplNetwork } from 'xpring-common-js'
+import IssuedCurrencyClient from '../../src/XRP/issued-currency-client'
+import { FakeXRPNetworkClient } from './fakes/fake-xrp-network-client'
+import 'mocha'
+
+const fakeSucceedingNetworkClient = new FakeXRPNetworkClient()
+// const fakeErroringNetworkClient = new FakeXRPNetworkClient(
+//   FakeXRPNetworkClientResponses.defaultErrorResponses,
+// )
```
Comment (tedkalaw) [inline test/XRP/issued-currency-client.test.ts:9]: what's this for

File: test/XRP/issued-currency-client.test.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+describe('Default XRP Client', function (): void {
+  it('Dummy description', function (): void {
+    // GIVEN an IssuedCurrencyClient.
+    const issuedCurrencyClient = new IssuedCurrencyClient(
+      fakeSucceedingNetworkClient,
+      XrplNetwork.Test,
+    )
+
+    // WHEN the compiler is told to shut up THEN everything is fine.
```
Comment (tedkalaw) [inline test/XRP/issued-currency-client.test.ts:19]: lol

**Review by tedkalaw — APPROVED** [review]
seems reasonable...would be down to silence eslint here intstead. let's be diligent about figuring out if we need the extra level of indirection. my sense is we'll be ok here since we get dependency injection through the network client maybe

File: test/XRP/issued-currency-client.test.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import IssuedCurrencyClient from '../../src/XRP/issued-currency-client'
+import { FakeXRPNetworkClient } from './fakes/fake-xrp-network-client'
+import 'mocha'
+
+const fakeSucceedingNetworkClient = new FakeXRPNetworkClient()
+// const fakeErroringNetworkClient = new FakeXRPNetworkClient(
+//   FakeXRPNetworkClientResponses.defaultErrorResponses,
+// )
+
+describe('Default XRP Client', function (): void {
```
Comment (mvadari) [inline test/XRP/issued-currency-client.test.ts:11]: Should be renamed

**Review by mvadari — APPROVED** [review]

File: test/XRP/issued-currency-client.test.ts (hunk 4 above)
Comment (amiecorso) [reply test/XRP/issued-currency-client.test.ts:11]: ah thank you

File: src/XRP/issued-currency-client.ts (hunk 1 above)
Comment (amiecorso) [reply src/XRP/issued-currency-client.ts:49]: ah yeah good idea I'll just get around with ESLint so that I'm not merging garbage to master

Comment (amiecorso) [reply src/XRP/issued-currency-client.ts:49]: dude I can't figure out eslint so i'm just going with it and Mayukha can remove this in her PR

File: test/XRP/issued-currency-client.test.ts (hunk 2 above)
Comment (amiecorso) [reply test/XRP/issued-currency-client.test.ts:9]: that's going to be needed shortly for the unit tests so I left the infra in

File: src/XRP/issued-currency-client.ts (hunk 1 above)
Comment (mvadari) [reply src/XRP/issued-currency-client.ts:49]: I can't remove it entirely :(, just one of the lines (I don't use the network client at all)

**Pull request merged**